\section{Introduction}

An ontology consists of a set of terms and their relations that represent a specific domain of knowledge. Ontologies are created and maintained by knowledge domain experts, and are used as computer-readable taxonomies by software tools intended to support knowledge management activities in that domain.

The complexity of some of the ontologies in current use, as well as the complexity of handling semantic metadata that annotate third party resources with ontology terms (e.g. as described in~\cite{BHW+11}), has generated considerable demand for effective visualization in the design, authoring, navigation and management of (i) ontology-based knowledge and (ii) semantic metadata that make use of ontologies.

In response to the above demand, a number of generic ontology visualization tools have been developed to assist knowledge acquisition, browsing and maintenance of ontologies~\cite{KHL+07}. Such tools, however, put a considerable and unrealistic demand on the users’ familiarity and expertise in both (i) semantic web technologies and (ii) the design principles of specific types of ontologies.

The domain of biomedical physiology is a case in point. Physiology experts deal with complex biophysical operations, across multiple spatial and temporal scales, that represent the transfer of energy from one form to another and/or from one anatomical location to another. Different kinds of descriptions of these biophysical operations are produced by different disciplines in biomedicine. For instance, (i) a medical doctor may describe the mechanism by which a stone in the ureter causes damage in the kidney; (ii) a pharmacologist may depict the process by which a drug transits from gut to the lung where it reduces inflammation; (iii) a molecular geneticist may trace the anatomical distribution of the expression of particular gene to understand the cause of a skeletal malformation; and, (iv) a bioengineer may build a mathematical model to quantify the effect of hormone production by the small intestine on the production of bile by the liver.

To ensure consistency, and semantic interoperability, of the above diverse accounts of physiology processes, the biomedical community is investing considerable effort in openly building ontologies for the annotation, and consequent semantic management, of data resources associated with these accounts.

In this paper, we present an integrated web-based environment for biomedical experts to navigate through schematically represented body tissues, browse, analyze and extend associated data resources and reason about multi-scale processes in selected regions of human body at any level of detail: from body parts and organs, to cells, molecules, proteins, and gene expressions. More specifically, we present a tool prototype that visualizes body parts, cardiovascular and neural connections, mathematical models of ongoing processes, static and dynamic 3D models, protein structures, etc.

First, we overview biomedical data resources we aim at providing access to and outline several usage scenarios that motivate our work (Section~\ref{sect:motivation}). Then we discuss data representation and visualization methods used to arrange and display relevant resources (Section~\ref{sect:visualization}). We describe a graphical tool prototype (Section~\ref{sect:implementation}).
Furthermore, we overview related efforts and techniques (Section~\ref{sect:relatedWork}).
Finally, we conclude the paper and discuss future work (Section~\ref{sect:conclusions}).
