%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Introduction}                                                                   %
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

Knowledge of physiology is extensive and complex. To provide software support for using and
manipulating physiology data, formalization of the knowledge is required. An \emph{ontology}
consists of a set of terms, and their relations, representing a specific domain of knowledge. They
are created and maintained by knowledge domain experts, and are used as computer-readable taxonomies
by software tools to support knowledge management activities in that domain. When knowledge is
formalised in this way, it is possible to record explicit descriptions of data elements in the
relevant domain using ontologies; this is the process of semantic annotation or the generation of
\emph{semantic metadata}.

%%%% Removed, as it is redundant and we have to drop a page
%
%The complexity of some of the ontologies in current use, as well as the complexity of handling
%semantic metadata that annotate third party resources with ontology terms (e.g. as described
%in~\cite{BHW+11}), has generated considerable demand for effective visualization in the design,
%authoring, navigation and management of (i) ontology-based knowledge and (ii) semantic metadata
%based on ontologies.

For example, physiology experts deal with complex biophysical operations across multiple spatial and
temporal scales, which they represent in terms of the transfer of energy from one form to another
and/or from one anatomical location to another. Different kinds of descriptions of these biophysical
operations are produced by different disciplines in biomedicine. For instance, (i)~a medical doctor
may describe the mechanism by which a stone in the ureter causes damage in the kidney; (ii)~a
pharmacologist may depict the process by which a drug absorbed from gut transits to the hip joints
where it reduces inflammation; (iii)~a molecular geneticist may trace the anatomical distribution of
the expression of a particular gene to understand the cause of a skeletal malformation; and, (iv)~a
bio-engineer may build a mathematical model to quantify the effect of hormone production by the
small intestine on the production of bile by the liver. These descriptions take diverse forms,
ranging from images and free text (e.g., journal papers) to models bearing well-defined data (e.g.,
from clinical trials) or sets of mathematical equations (which might be used as input for a
simulation tool).

The physiology community is investing
considerable effort in building ontologies for the annotation and semantic management of such
resources. For example, a number of reference ontologies have been created to represent gene
products~\cite{Bla+13}, chemical entities~\cite{HMD+13}, cells~\cite{BRA05} and gross
anatomy~\cite{RM03}. Together, these ontologies consist of hundreds of thousands of terms, such that
the volume of semantic metadata arising from resource annotation is considerable.

Unfortunately,
conventional technology for the visualization and management of ontologies and metadata is not
usefully accessible to
physiology experts, as they involve unfamiliar, abstract technicalities.
%%
Having to become technically proficient with such technology is a burden few physiology experts can
bear without losing touch with their long term goals.
%%
Rather, domain experts should be able to manage data based on a familiar perspective, in which its
meaning is made explicit in terms of the expert's own knowledge; a long standing challenge in
knowledge engineering.

%%%% TODO: integrate into above?
%
%In response to the above demand, a number of generic ontology visualization tools have been
%developed to assist knowledge acquisition, browsing and maintenance of ontologies~\cite{KHL+07}.
%Such tools, however, put considerable and unrealistic demands on the users' familiarity and
%expertise in both semantic web technologies and the design principles of ontologies. It is unlikely
%that a user with \emph{domain} expertise also has expertise in the \emph{technologies} used to
%manage ontologies.

%%%% Removed; TODO: move some of this to related work?
%
%The present work builds on past efforts on RICORDO~\cite{BHW+11}. Those efforts have led to the
%design of backend software for the management of physiology metadata, including physiology models,
%genomics data, and radiology and histology images. This software comprises four notionally separable
%components: (a) expert knowledge, (b) a data repository, (c) a metadata repository and (d) an
%ontology knowledge base able to perform reasoning over ontology relationships.
%The ApiNATOMY effort has emerged to provide an intuitive graphical frontend to RICORDO.

In this paper, we present ApiNATOMY, a web-based environment that allows physiology experts to
navigate through circuitboard visualizations of body components, and their cardiovascular and neural
connections, across different scales. It supports a plugin infrastructure to overlay
graphical renderings of organs, neurons and gene products on these schematics, in support of
biomedical knowledge management use cases discussed in the next section.

The remainder of the paper is structured as follows: \cref{sect:motivation} gives an overview of the
\mbox{ontology-,} metadata- and data-resources that we focused on for the ApiNATOMY prototype, and
outline key use-case scenarios that motivate our work. \Cref{sect:visualization,sect:visualization2}
then discuss the visualization techniques we applied to arrange and display those resources.
\Cref{sect:implementation} provides some insight into the implementation of the prototype. Finally,
\cref{sect:relatedWork,sect:conclusions} offer an overview of related efforts in the field, and
conclude the paper with a discussion of the anticipated implications of our tool, as well as planned
future work.















